includes/templates/template_default/jscript/zen_addr_pulldowns.php

```diff
@@ -11,58 +11,77 @@
  */
 // -----
-// If the site has NOT enabled states to be displayed as dropdowns, nothing to be done here!
+// If the site does NOT require a 'State' entry in an address, nothing to be done here!
 //
-if (ACCOUNT_STATE_DRAW_INITIAL_DROPDOWN !== 'true') {
+if (ACCOUNT_STATE !== 'true') {
     return;
 }
-?>
-<script>
-jQuery(document).ready(function() {
-<?php
+
 // -----
-// Create a JSON-encoded array of countries-to-zones for use by the jQuery section.
+// If the current site is using the state dropdowns, create the array that identifies the various
+// zones for the currently-active countries.  When state dropdowns aren't being used, the
+// empty $c2z array will 'instruct' the jQuery section to presume that no countries have
+// associated zones.
 //
-$countries = $db->Execute(
-    "SELECT DISTINCT zone_country_id
-       FROM " . TABLE_ZONES . "
-            INNER JOIN " . TABLE_COUNTRIES . "
-                ON countries_id = zone_country_id
-               AND status = 1
-   ORDER BY zone_country_id"
-);
-
 $c2z = [];
-foreach ($countries as $next_country) {
-    $current_country_id = $next_country['zone_country_id'];
-    $c2z[$current_country_id] = [];
-
-    $states = $db->Execute(
-        "SELECT zone_name, zone_id
+if (ACCOUNT_STATE_DRAW_INITIAL_DROPDOWN === 'true') {
+    // -----
+    // If the current site has at least one country enabled that uses zones, a JSON-encoded array of
+    // countries-to-zones will be created for use by the jQuery section.
+    //
+    $countries = $db->Execute(
+        "SELECT DISTINCT zone_country_id
            FROM " . TABLE_ZONES . "
-          WHERE zone_country_id = $current_country_id
-       ORDER BY zone_name"
+                INNER JOIN " . TABLE_COUNTRIES . "
+                    ON countries_id = zone_country_id
+                   AND status = 1
+       ORDER BY zone_country_id"
     );
-    foreach ($states as $next_state) {
-        $c2z[$current_country_id][$next_state['zone_id']] = $next_state['zone_name'];
-    }
-}
+    foreach ($countries as $next_country) {
+        $current_country_id = $next_country['zone_country_id'];
+        $c2z[$current_country_id] = [];
 
-if (count($c2z) !== 0) {
-    echo '    var country_zones = \'' . addslashes(json_encode($c2z)) . '\';' . PHP_EOL;
+        $states = zen_get_country_zones($current_country_id);
+        foreach ($states as $next_state) {
+            $c2z[$current_country_id][$next_state['id']] = $next_state['text'];
+        }
+    }
 }
-
+?>
+<script>
+jQuery(document).ready(function() {
+    var country_zones = '<?php echo addslashes(json_encode($c2z)); ?>';
+<?php
 // -----
-// Initialize the display for the dropdown vs. hand-entry of the state fields.  If the initially-selected
-// country doesn't have zones, the dropdown will contain only 1 element ('Type a choice below ...').
+// Notes:
+//
+// 1. The '#stBreak' <br> is also never needed/wanted since it will 'throw' the state-field input underneath
```

Comment:
If '#stBreak' always should be hidden, why not remove it from the template file?

Reply:
It's `legacy` with existing add-on templates already using it.  Keeping it documented will give the base the opportunity to remove it as 'best practices' in the future.

Reply:
See separate issue created, as identified below.